packages/ra-ui-materialui/src/list/filter/FilterForm.tsx

```diff
@@ -237,9 +237,15 @@ const getInputValue = (
     key: string,
     filterValues: Record<string, any>
 ) => {
+    if (formValues[key] === undefined || formValues[key] === null) {
```

Comment:
You get `null` when a DateInput/DateTimeInput  clear button is click by the user